### ISSUE TITLE
Check build.cmd arguments before we start doing actual work

### ISSUE THREAD

**Issue by carlossanlop** [dotnet/runtime#62535]
If you have a fresh repo (no artifacts) and you run `./build.cmd` with an argument that has a typo or is misused, the script will perform some verifications, including downloading the sdk zip file and installing it. If you have a bad internet connection, this could take a long time. Then after that process finishes, you get the error telling you that you wrote your command wrong.

Can we explore the possibility of changing the order of the initial tasks of `build.cmd`, particularly checking arguments, to ensure the script ends quickly if there's misusage?

**Comment by ivdiazsa** [comment]
I agree this would help a lot. It's really annoying to leave a build running, go get coffee or lunch, and come back to see it failed because the command was wrong. This gets worse if you're on Windows, as it takes even longer than Linux to finish running checks and downloading the SDK, and therefore longer to fail before attempting the build.

I was looking at this issue just now, and some options might require more validation work than others. Here are the observations I came upon.

* Parameters defined as `ValidateSet` in `eng/build.ps1` are safe, as those fail right away. We also implemented something similar in `eng/build.sh`, so MacOS and Linux are safe as well.

* Unknown flags by mistake to the scripts, usually in the form of typos like `--suibset` for example, are passed as potential MSBuild arguments, and therefore can't fail until the SDK is downloaded and active. For those ones, I have a potential solution:
  * Do we ever pass stuff to MSBuild through the build scripts, without prefixing it with `-t:` or `-p:`? If not, we could add a check to verify that all unknown flags are prefixed. That way, we don't have to wait for the SDK.

* We have some open string options, such as `--framework` and `--verbosity`. I believe we could use some sort of regexp for *framework*, and we could convert *verbosity* to a defined set, as I believe we only have like 5 or 6 valid values for this flag.

What do you think @carlossanlop @jkoritzinsky?
